central-service/services/room.go

```diff
@@ -111,7 +113,16 @@ func (service *RoomService) JoinRoom(roomID string, userID string) (*models.Room
 			}
 		}
 	}
-	// TODO: notify RTC
+	// RTCClient is nil in test cases
+	if service.rtcClient != nil {
+		joinRoom := requests.JoinRoomRequest{
+			UserHandle: userID,
+			RoomID:     roomID,
+		}
+		if _, err = service.rtcClient.SendMessage("join-room", joinRoom); err != nil {
+			log.Printf("Error sending join-room message: %v", err)
```

Comment:
Best to return something here, otherwise, the JoinRoom function will always return nil err.

Reply:
NIT: (can re-write)
``` go
// RTCClient is nil in test cases
 if service.rtcClient == nil {
    return nil, errors.New("rtc client is nil")
 }
 
joinRoom := requests.JoinRoomRequest{
    UserHandle: userID,
    RoomID:     roomID,
}

if _, err = service.rtcClient.SendMessage("join-room", joinRoom); err != nil {
    log.Printf("error sending join-room message: %v", err)
    return nil, err
}